src/main/java/com/fulcrumgenomics/util/PickLongIndices.scala

```diff
@@ -0,0 +1,291 @@
+  private[util] val AdapterRegex = Pattern.compile("^[ACGT]*N+[ACGT]*$")
+  adapters.map(_.toUpperCase).filterNot(AdapterRegex.matcher(_).matches()) match {
+    case Seq() => Unit
+    case xs    => invalid(s"Adapters must be all [ACGT] with a single contiguous block of Ns where the index goes: ${xs}")
+  }
+
+  /** The main method! */
+  override def execute(): Unit = {
+    val picks = pickIndices
+    logger.info(s"Picked ${picks.size} indices.")
+    writeOutput(picks, output)
+  }
+
+  /** Picks a set of indices. */
+  private[util] def pickIndices: IndexSet = {
+    val indices = new IndexSet
+    ExistingIndices.foreach(i => indices += i)
+
+    var ok = true
+    while (ok && indices.size < numberOfIndices) {
+      val pick = nextIndex(indices, this.length, this.editDistance, this.attempts)
+      pick.foreach(p => indices += p)
+      ok = pick.nonEmpty
+    }
+
+    indices
+  }
+
+  /**
+    * Attempts to generate the next index that can be added to the set without violating any
+    * of the constraints.  Will evaluate 'attempts' indices which would be acceptable independently
+    * to see if they can be added.
+    *
+    * @return Some(Index) if an index can be found within the number of attempts, otherwise None
+    */
+  private def nextIndex(picks: IndexSet, length: Int, minEdits: Int, attempts: Int): Option[Index] = {
+    var pick: Option[Index] = None
+    var attempt = 0
+    while (attempt < attempts && pick.isEmpty) {
+      val index = randomAcceptableIndex(length)
+      val string = new String(index)
+
+      // structure checking (worstDeltaG) is done here instead of in randomAcceptableIndex because,
+      // even though it's a per-index property, it's much slower than all the other checks, and
+      // when making many attempts to find a suitable next index, the time spent structure checking
+      // all the indices we're going to discard for too-few-edits anyway explodes.
+      if ( !picks.contains(string) &&
+           (allowReverses || !picks.contains(string.reverse)) &&
+           (allowReverseComplements || !picks.contains(SequenceUtil.reverseComplement(string))) &&
+           picks.forall(p => mismatches(index, p) >= minEdits) &&
+           worstDeltaG(index).forall(_ >= minDeltaG)) {
+        pick = Some(index)
+      }
+      attempt += 1
+    }
+
+    pick
+  }
+
+  /** Generates the next random index which is acceptable as a standalone index, prior to any secondary structure checking. */
+  @tailrec private def randomAcceptableIndex(length: Int): Index = {
+    val index  = randomIndex(length)
+    val string = new String(index)
+    val gc = SequenceUtil.calculateGc(index)
+
+    if (gc >= minGc && gc <= maxGc &&
+        homopolymerLengthOk(index, maxHomopolymer) &&
+        !AvoidKmers.contains(string) &&
+        (allowPalindromes || !isPalindrome(index))
+    ) {
+      index
+    }
+    else {
+      randomAcceptableIndex(length)
```

Comment:
I think that's possible, yes, if you fed in pathalogical parameters.  But it would be super unlikely.